### PR TITLE
git-magic: add page

### PR DESCRIPTION
**Pull request by Sadeeed**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- **Version of the command being documented (if known):**

Issue: #5137

### CODE REVIEW

File: pages/common/git-magic.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+> Automate add, commit, and push routines.
+> Part of `git-extras`.
+> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-magic>.
+
+- Commit changes with a generated message:
+
+`git magic`
+
+- Add untracked files and commit changes with a generated message:
```
Comment (reinhart1010) [inline pages/common/git-magic.md:11]:
```suggestion
- [a]dd untracked files and commit changes with a generated message:
```

`[a]` (case-sensitive) is a command mnemonic. Learn more on our [style guide](https://github.com/tldr-pages/tldr/blob/main/contributing-guides/style-guide.md).

File: pages/common/git-magic.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+`git magic -a`
+
+- Commit changes with a custom message:
```
Comment (reinhart1010) [inline pages/common/git-magic.md:15]:
```suggestion
- Commit changes with a custom [m]essage:
```

File: pages/common/git-magic.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+`git magic -m "{{custom_commit_message}}"`
+
+- Edit the commit message before committing:
```
Comment (reinhart1010) [inline pages/common/git-magic.md:19]:
```suggestion
- [e]dit the commit [m]essage before committing:
```

File: pages/common/git-magic.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+`git magic -em "{{custom_commit_message}}"`
+
+- Commit changes and push to remote:
```
Comment (reinhart1010) [inline pages/common/git-magic.md:23]:
```suggestion
- Commit changes and [p]ush to remote:
```

File: pages/common/git-magic.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+`git magic -p`
+
+- Commit changes with a force push to remote:
```
Comment (reinhart1010) [inline pages/common/git-magic.md:27]:
```suggestion
- Commit changes with a [f]orce [p]ush to remote:
```

**Review by reinhart1010 — CHANGES_REQUESTED** [review]
Thanks for adding a new page! Here are some of my notes to review.

File: pages/common/git-magic.md (hunk 1 above)
Comment (Sadeeed) [reply pages/common/git-magic.md:11]: I can't find anything about it in the style guide but I'll keep that in mind for the future. Thanks

**Review by reinhart1010 — APPROVED** [review]
Thanks!

**Pull request merged**